botwo/botwo.py

```diff
@@ -0,0 +1,188 @@
+import fnmatch
+
+SPECIAL_METHODS = {"get_paginator", "delete_objects"}
+COPY_METHODS = {"copy", "copy_object", "copy_upload_part"}
+
+
+def client(client_mapping, config):
+    botwo = Botwo()
+    return botwo.create_client_class(client_mapping, config)
+
+
+class Paginatwo(object):
+
+    def __init__(self, mapping, kwargs, client1, client2, client1_name, client2_name):
+        self.mapping = mapping
+        self.client1_name = client1_name
+        self.client2_name = client2_name
+        self.paginator1 = client1.get_paginator(kwargs["operation_name"])
+        self.paginator2 = client2.get_paginator(kwargs["operation_name"])
+
+    def paginate(self, **kwargs):
+        paginator_to_call = self.paginator1
+        for map in self.mapping:
+            if fnmatch.fnmatch(kwargs["Bucket"], map["bucket_name"]):
+                if map.get("client"):
+                    if map.get("client") == self.client2_name:
+                        paginator_to_call = self.paginator2
+                    elif map.get("client") == self.client1_name:
+                        paginator_to_call = self.paginator1
+                    else:
+                        raise ValueError("Invalid client name " + map.get("client"))
+                else:
+                    raise ValueError("client isn't provided")
+
+                if map.get("mapped_bucket_name"):
+                    kwargs["Bucket"] = map.get("mapped_bucket_name")
+                if kwargs.get("Prefix"):
+                    if map.get("key_prefix"):
+                        kwargs["Prefix"] = map.get("keu_prefix") + kwargs["Prefix"]
+                break
+        return getattr(paginator_to_call, "paginate")(**kwargs)
+
+
+class Botwo(object):
+    def __init__(self):
+        self.client1 = None
+        self.client2 = None
+        self.default_client = None
+        self.client1_name = None
+        self.client2_name = None
+        self.mapping = None
+        self.meta = None
+
+    def create_client_class(self, client_mapping, config):
+        if type(client_mapping) is not dict:
+            raise TypeError("Invalid client mapping type: " + str(type(client_mapping)) + " expected dict")
+        if len(client_mapping) != 2:
+            raise TypeError("two clients is expected")
+
+        names = list(client_mapping)
+        self.client1_name = names[0]
+        self.client1 = client_mapping.get(names[0])
+        self.client2_name = names[1]
+        self.client2 = client_mapping.get(names[1])
+
+        self.mapping = self._parse_config(config)
+        class_attributes = self._create_methods()
+
+        bases = [Botwo]
+        cls = type("s3", tuple(bases), class_attributes)
```

Comment:
IIUC this creates a new type that is named "s3".  That is confusing (at least to me), because this class is called `Botwo` and (I think...) it pretends to be an object of type [`S3.Client`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client).